plugins/org.yakindu.sct.doc.user/src/user-guide/embedded_systems_integration_guide.textile

```diff
@@ -0,0 +1,281 @@
+On a pure bare-metal implementation there is no such thing as elapsed time. Everything depends on cycles. 
+
+Imagine a microcontroller running with a clock of 16MHz. Running one instruction needs 1/16MHz = 625ns. A delay() function 1000000 would need 6,25 ms.A first approach updating the timer could be:
+
+bc.. 
+void main() {
+	while(true) {
+		delay(1000000);
+		updateTimer(6,25);
+		runCycle();
+	}
+}
+p. 
+
+But using this implementations got two big drawbacks:
+* the delay function is blocking
+* the execution time of the updateTimer and runCycle are not taking into account
+
+A much more precise design is using timer interrupts, which can generate a periodic interrupt ever x cycles:
+
+bc.. 
+void handleTimer() {
+	if(updateTimerFlag) {
+		updateTimer(TIMER_TICK_MS);
+		updateTimerFlag = false;
+	}
+}
+
+#define TIMER_TICK_MS 32 // 32 ms for example
+Timer_ISR{ 
+	updateTimerFlag = true;
+}
+p. 
+
+Some microcontrollers, like the Arduino, support functions like millis(), which are in fact using a similar method, but storing the elapsed time since starting the device. This time can be used to determine the elapsed time between two cycles in a loop:
+
+bc.. 
+void loop() {
+	current_millis = millis();
+	updateTimer(current_millis - last_cycle_time);
+	statemachine_runCycle();
+	last_cycle_time = current_millis;
+}
+
+p. 
+
+For prototyping, this implementation can be used, but it should be considered that the timer will overflow after approximately 50 days. That's why we highly recommend a implementation with interrupts as described before.
+
+h3(#handle-out-events). Handle Out Events
+
+Out events raised by the state machine can be checked by using the state machine's API. If an out event gets raised its returning value is true, until the next runCycle is executed. This is why the out events should be handled <b>after</b> every runCycle call. They can be used to wire up different actuators. Mapping values, e.g. integers, to the out events, is also possible.
+
+bc.. 
+void handleOutEvents() {
+	if(statemachine_israised_outEvent1()) {
+		controlActuator1();
+	}
+	if(statemachine_israised_outEvent2()) {
+		controlActuator2(statemachine_get_outEvent2_value());
+	}
+}
+p. 
+
+h3(#execution-as-state-machine). Execution As State Machine
+
+This is a expanded view on the implementation, of how it can be realized as a state machine.
+(needs to be updated)!
+
+!(large)images/images_embedded_systems_integration_guide/platform_integration.png(Complete execution shown as State Machine)!
+
+p=. Complete execution shown as State Machine
+
```

Comment:
it is already described and what is describe is a meta state machine. I won't go for too much meta here ...

Reply:
:+1: